src/Spark.Mongo/Search/Searcher/CriteriaMongoExtensions.cs

```diff
@@ -418,7 +422,7 @@ private static FilterDefinition<BsonDocument> UriQuery(String parameterName, Ope
                 default:
                     throw new ArgumentException(string.Format("Invalid modifier {0} on Uri parameter {1}", modifier, parameterName));
             }
-            return StringQuery(parameterName, optor, localModifier, operand2);
+            return StringQuery(parameterName, optor, localModifier, operand);
```

Comment:
Instead of manipulating the value of one of the in parameters I would rather we do it like this
```suggestion
            return StringQuery(parameterName, optor, localModifier, new UntypedValue(UriUtil.NormalizeUri(operand.ToUnescapedString())));
```